src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,63 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const _ = require("lodash");
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["msg", "say", "query"];
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = cmd === "say" ? chan.name : args.shift();
+
+	if (cmd === "query") {
+		if (args.length === 0 || target.length === 0) {
```

Comment:
I think due to the args.shift above, this will break opening query windows without any message. `/query user`.

Reply:
You were correct, this broke direct message.